ToolBox/USTreasuryYieldCurve/USTreasuryYieldCurveConverter.cs

```diff
@@ -0,0 +1,120 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using Newtonsoft.Json;
+using QuantConnect.Logging;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+
+namespace QuantConnect.ToolBox.USTreasuryYieldCurve
+{
+    public class USTreasuryYieldCurveConverter
+    {
+        private readonly DirectoryInfo _sourceDirectory;
+        private readonly DirectoryInfo _destinationDirectory;
+
+        public USTreasuryYieldCurveConverter(string sourceDirectory, string destinationDirectory)
+        {
+            _sourceDirectory = new DirectoryInfo(sourceDirectory);
+            _destinationDirectory = new DirectoryInfo(destinationDirectory);
+            _destinationDirectory.Create();
+        }
+
+        /// <summary>
+        /// Converts the U.S. Treasury yield curve data to CSV format
+        /// </summary>
+        public void Convert()
+        {
+            Log.Trace("USTreasuryYieldCurveRateConverter.Convert(): Begin converting U.S. Treasury yield curve rate data");
+
+            var rawFile = new FileInfo(Path.Combine(_sourceDirectory.FullName, "yieldcurverates.xml"));
+            var finalPath = new FileInfo(Path.Combine(_destinationDirectory.FullName, "yieldcurverates.csv"));
+
+            if (!rawFile.Exists)
+            {
+                throw new FileNotFoundException($"Failed to find yield curve rates file: {rawFile.FullName}");
+            }
+
+            using (var stream = rawFile.OpenText())
+            {
+                Log.Trace("USTreasuryYieldCurveConverter.Convert(): Begin deserialization of raw XML data");
+                var xmlData = (feed) new XmlSerializer(typeof(feed))
+                    .Deserialize(stream);
+
+                // I don't think this should happen, but let's make sure before we work with the type
+                if (xmlData == null)
+                {
+                    throw new NullReferenceException("XML data is null. Perhaps we're deserializing the wrong XML data?");
```

Comment:
`NullReferenceException` is typically viewed as a framework reserved exception (along w/ others such as `AggregateException`) -- perhaps a better exception type is `InvalidOperationException` which is used when an operation is not valid given the current state of the object.